From 97afdc42a3cd50de94d824fab06b8c4a13a20c29 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 3 May 2015 12:58:40 +0200
Subject: [PATCH 05/12] Enhance get_repo_list function

Adds a PEP257 docstring and use layman API to provide a description to
repositories.
---
 backends/portage/portageBackend.py | 24 +++++++++++++++---------
 1 file changed, 15 insertions(+), 9 deletions(-)

diff --git a/backends/portage/portageBackend.py b/backends/portage/portageBackend.py
index f37542e..ddeca0d 100755
--- a/backends/portage/portageBackend.py
+++ b/backends/portage/portageBackend.py
@@ -1016,10 +1016,14 @@ class PackageKitPortageBackend(PackageKitPortageMixin, PackageKitBaseBackend):
         self.percentage(100)
 
     def get_repo_list(self, filters):
-        # NOTES:
-        # use layman API
-        # returns only official and supported repositories
-        # and creates a dummy repo for portage tree
+        """ Get list of repository.
+
+        Get the list of repository tagged as official and supported by current
+        setup of layman.
+
+        Adds a dummy entry for gentoo-x86 official tree even though it appears
+        in layman's listing nowadays.
+        """
         self.status(STATUS_INFO)
         self.allow_cancel(True)
         self.percentage(None)
@@ -1033,11 +1037,13 @@ class PackageKitPortageBackend(PackageKitPortageMixin, PackageKitBaseBackend):
         self.repo_detail('gentoo', 'Gentoo Portage tree', True)
 
         if FILTER_NOT_DEVELOPMENT not in filters:
-            for o in available_layman_db.overlays.keys():
-                if available_layman_db.overlays[o].is_official() \
-                        and available_layman_db.overlays[o].is_supported():
-                    self.repo_detail(o, o,
-                            self._is_repo_enabled(installed_layman_db, o))
+            for repo_name, overlay in available_layman_db.overlays.items():
+                if overlay.is_official() and overlay.is_supported():
+                    self.repo_detail(
+                        repo_name,
+                        overlay.name,
+                        self._is_repo_enabled(installed_layman_db, repo_name)
+                    )
 
     def required_by(self, filters, pkgs, recursive):
         # TODO: manage non-installed package
-- 
2.3.6

